File: services/api/src/libs/chunk.ts
export function chunkText(text: string, chunkSize = 1000, overlap = 200): string[] {
    if (chunkSize <= overlap) throw new Error("chunkSize must be greater than overlap");

    const chunks: string[] = [];
    let start = 0;

    while (start < text.length) {
        const end = Math.min(start + chunkSize, text.length);
        const part = text.slice(start, end).trim();
        if (part) chunks.push(part);

        if (end === text.length) break;
        start = end - overlap;
    }

    return chunks;
}

File: services/api_py/qa/chunk.py
from __future__ import annotations


def chunk_text(text: str, chunk_size: int = 1000, overlap: int = 200) -> list[str]:
    if chunk_size <= overlap:
        raise ValueError("chunkSize must be greater than overlap")

    chunks: list[str] = []
    start = 0

    while start < len(text):
        end = min(start + chunk_size, len(text))
        part = text[start:end].strip()
        if part:
            chunks.append(part)

        if end == len(text):
            break
        start = end - overlap

    return chunks
